src/dsp_tools/dsp_tools.py

```diff
@@ -302,16 +316,65 @@ def _log_cli_arguments(parsed_args: argparse.Namespace) -> None:
     logger.info("*" * asterisk_count)
 
 
-def call_requested_action(
-    user_args: list[str],
-    parser: argparse.ArgumentParser,
-) -> bool:
+def _derive_sipi_url(
+    parsed_arguments: argparse.Namespace,
+    default_dsp_api_url: str,
+    default_sipi_url: str,
+) -> argparse.Namespace:
```

Comment:
Good point, thanks for pointing this out!

On the other hand, I want to have a clean `main()` method, that does nothing else than calling methods of the next abstraction level. So, ideally, I would introduce a new method that takes care of the string processing only. But then I still need this method here that takes care of introducing the strings into the Namespace, like this: 

```
def _get_strings(...)
    ...

def _derive_sipi_url(...)
    server, sipi = _get_strings(...)
    parsed_arguments.server = server
    parsed_arguments.sipi = sipi
    return parsed_arguments

def main()
    _derive_sipi_url(...)
```

While this would perhaps be the cleanest approach, I prefer leaving it as it is. But thanks for the Denkanstoss :-)

Reply:
I would generally go for the cleaner variant, wherever possible. But looking at the `dsp_tools.py` file again, I think the whole file needs refactoring anyway, so this is probably arguing about details.